packages/knex/src/query/QueryBuilderHelper.ts

```diff
@@ -409,7 +409,18 @@ export class QueryBuilderHelper {
       return void qb[m](this.knex.raw(`(${this.subQueries[key]})`), replacement, value[op]);
     }
 
-    qb[m](this.mapper(key, type, undefined, null), replacement, value[op]);
+    if (op === '$fulltext') {
+      const entityMetadata = this.metadata.get(this.entityName);
```

Comment:
lets use just `meta`, as in the whole codebase
```suggestion
      const meta = this.metadata.get(this.entityName);
```